src/main/java/org/mitre/synthea/helpers/Utilities.java

```diff
@@ -338,6 +345,46 @@ private static String getSyntheaVersion() {
     return version;
   }
 
+  /**
+   * Parse a range of Synthea timestamps (milliseconds since the epoch) from a String. The string
+   * can be in one of two formats:
+   * "milliseconds start-milliseconds end" example: "1599264000000-1627948800000"
+   * or
+   * "ISO date start-ISO date end" example: "2020-09-05-2021-08-03"
+   * @param range String containing the range
+   * @return A Range with the min and max set to Synthea timestamps, longs containing milliseconds
+   *     since the epoch
+   * @throws IllegalArgumentException If the string is not in one of the expected formats
+   */
+  public static Range<Long> parseDateRange(String range) throws IllegalArgumentException {
+    if (!range.contains("-")
+        || range.substring(0, range.indexOf("-")).length() < 1
+        || range.substring(range.indexOf("-") + 1).length() < 1) {
+      throw new IllegalArgumentException("Time range format error. Expect low-high. Found '"
+              + range + "'");
+    }
+
+    Pattern dateRangeRegex =
+        Pattern.compile("^(\\d{4}\\-\\d{2}\\-\\d{2})\\-(\\d{4}\\-\\d{2}\\-\\d{2})$");
+
+    Matcher matcher = dateRangeRegex.matcher(range);
+
+    Range<Long> parsedRange;
+    if (matcher.matches()) {
+      parsedRange = Range.between(
+        LocalDate.parse(matcher.group(1), DateTimeFormatter.ISO_LOCAL_DATE).atStartOfDay()
+            .toInstant(ZoneOffset.UTC).toEpochMilli(),
+        LocalDate.parse(matcher.group(2), DateTimeFormatter.ISO_LOCAL_DATE).atStartOfDay()
```

Comment:
I wonder if it would be better for the upper bound to be the end of the day so that the range is fully inclusive instead of inclusive of the start and exclusive of the end? Either way, the Javadoc should specify inclusive vs exclusive.